app/components/sections/view.rb

```diff
@@ -102,7 +102,7 @@ def path
           review: "trainee_degrees_confirm_path",
         },
         course_details: {
-          incomplete: "edit_trainee_course_details_path",
+          incomplete: "edit_trainee_publish_course_details_path",
```

Comment:
If you look in `TaskListHelper`, there's actually a bit of logic that decides what this path should be on the `review-draft` page. Shouldn't we be using the same logic here? (see `def path_for_course_details`)

For example, we shouldn't be heading down the publish_course_details_path for an assessment only trainee - they are never on publish courses. In fact, if you try on your review app for an AO trainee, it errors:

<img width="832" alt="Screenshot 2021-10-06 at 11 27 32" src="https://user-images.githubusercontent.com/18436946/136186864-9c580f96-c7e7-4ed8-a2de-ba8b7685b738.png">